[profiler][refactor] Refactor the usage of legacy profiler implementation

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61931

Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

Commit:
[profiler][refactor] Remove usage of legacy profiler implementation

Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

kimishpatel commented on torch/csrc/autograd/profiler_kineto.cpp:
```diff
@@ -112,43 +128,82 @@ struct KinetoThreadLocalState : public ProfilerThreadLocalState {
     if (config_.profile_memory && config_.state != ProfilerState::Disabled) {
       std::lock_guard<std::mutex> guard(state_mutex_);
+      auto start_time = getTimeUs();
+#ifdef USE_KINETO
```
So now memory related event is added as a `libkineto::GenericTraceActivity` meaning this feature is available only if USE_KINETO

kimishpatel commented on torch/csrc/autograd/init.cpp:
```diff
@@ -202,20 +211,22 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject *unused) {
     .def("events", &ProfilerResult::events)
-    .def("save", &ProfilerResult::save);
+#ifdef USE_KINETO
+    .def("save", &ProfilerResult::save)
```
Also chrome trace saving is available only if you use KINETO?

ilia-cher replied:
if kineto is not available we use a python implementation for trace export (in profiler_util.py)

kimishpatel replied:
Ok thats good.

kimishpatel commented on torch/csrc/autograd/init.cpp:
```diff
@@ -202,20 +211,22 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject *unused) {
 
   m.def("_add_metadata_json", [](const std::string& key, const std::string& value) {
 #ifdef USE_KINETO
       addMetadataJson(key, value);
```
Interesting. I did not realize that this function was available only with kineto.

ilia-cher replied:
yes, that's to save the extra data into the kineto events, to be saved later in the trace (by kineto); we don't support this in the non-kineto case; generally speaking, profiler without kineto has less capabilities, and that's expected; in almost all use cases profiler is built with kineto support, I'd recommend exploring this for the remaining cases as, e.g. mobile builds - cpu-only kineto library should not add too much to the build size

kimishpatel replied:
I hope that size burden is not too much. But profiling build wont be production build so we wont include kineto dependency in production anyway, but it would still be nice to optionally include cpu only kineto lib, or at least for internal builds we can have separate cpu only target to link against.

kimishpatel commented on torch/csrc/autograd/profiler_kineto.cpp:
```diff
@@ -112,43 +128,82 @@ struct KinetoThreadLocalState : public ProfilerThreadLocalState {
-      evt.activity(act)
-        .deviceType(device.type())
+      evt.name(kMemoryEventName)
+        .startUs(start_time)
```
Ignore my earlier comment about memory profiling available in kineto only.

kimishpatel reviewed (commented):
Mostly looks good to me. I left some small comments for clarification.

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

kimishpatel commented on torch/csrc/autograd/profiler_kineto.cpp:
```diff
@@ -321,8 +321,7 @@ void pushProfilingCallbacks() {
             } catch (const std::exception& e) {
-              C10_LOG_EVERY_N(WARNING, 1000) << "Failed to record CUDA event. "
-                                            << e.what();
+              LOG(WARNING) << "Failed to record CUDA event. " << e.what();
```
So this change logs every time unlike pervious? I assume this is ok because logging happens only in case of exceptions?

ilia-cher replied:
yes, unfortunately I had to avoid using LOG_EVERY_N because of the dependency on some old glog version which doesn't work with the newer clang in some of the internal builds

kimishpatel commented on torch/csrc/distributed/rpc/request_callback_no_python.cpp:
```diff
@@ -82,7 +83,7 @@ c10::intrusive_ptr<JitFuture> RequestCallbackNoPython::processMessage(
             // Initialize thread-local profiler state from process-global
             // profiler state.
-            ::torch::autograd::profiler::enableProfilerLegacy(
+            enableProfilerLegacy(
```
Will we eventually replace all these with kineto profiler?

kimishpatel reviewed (commented):
Looks good to me, but I think it will be good to have @gdankel review this as well.

Commit:
Update on "[profiler][refactor] Remove usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Refactor the usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).

Commit:
Update on "[profiler][refactor] Refactor the usage of legacy profiler implementation"


Summary:
This PR consolidates the profiling code around a new C++ implementation
(profiler_kineto.h/cpp) and uses it unconditionally from
torch.autograd.profiler/torch.profiler:
1. Always use profiler_kineto.h/cpp as the C++ implementation
2. Simplify profiler.py to remove unneeded parts depending on legacy
impl
3. Move some of the legacy logic into profiler_legacy.py (to be fully
deleted later)

Test Plan:
USE_KINETO=1 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v
USE_KINETO=0 USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake
python test/test_profiler.py -v

Reviewers: gdankel

Differential Revision: [D29801599](https://our.internmc.facebook.com/intern/diff/D29801599)

[ghstack-poisoned]

ilia-cher commented:
@ilia-cher has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801599).
